src/OICodeGen.h

```diff
@@ -114,8 +114,11 @@ class OICodeGen {
   Config config{};
   FuncGen funcGen;
 
-  using ContainerTypeMap =
-      std::pair<ContainerInfo, std::vector<drgn_qualified_type>>;
+  using RefSharedContainerInfo =
```

Comment:
What does a reference_wrapper of a shared_ptr provide? Just an optimisation to avoid bumping the refcount?

Reply:
Yep. I was aiming for `&` for just reading, `&shared_ptr` for reading and potentially copying, and `shared_ptr` for owning. In theory this is quicker in a multithreaded context but as we're currently single threaded it's not, so it was a bit of future proofing. In hindsight it was most of the work of this diff to make the really awkward reference-wrappers work...